### PR TITLE
Bluetooth: controller: Fix null pointer deference during adv data update

### LINKED ISSUE #28544: Null pointer dereference in ll_adv_aux_ad_data_set
**Describe the bug**
When updating advertising data with `bt_le_adv_update_data` I get error and warning message in log: "bt_hci_core: opcode 0x2037 status 0x0d". I'm not using extended advertising. 

I debugged the problem to be in `ll_adv_aux_ad_data_set` where controller updates advertising data and after that tries to start extended advertising even if it is not used, however it doesn't check `adv->lll.aux` for NULL.

**To Reproduce**
I'm running on custom board with modifications to zephyr, but I think the problem should be reproducible with any BT board as long as CONFIG_BT_EXT_ADV=y or even by looking at the source.

**Expected behavior**
`bt_le_adv_update_data` should return 0.

**Impact**
Fixed in my fork so no problem for me.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: gcc-arm-none-eabi-8-2018-q4-major
 - Commit 122bc38fad422226fc1fc62ef844afa035f0fefe + custom modifications

### PR DESCRIPTION
**Pull request by J-Montgomery**
Added a check for adv->lll.aux before starting extended advertising to
fix observed NULL pointer dereference when updating AD data of legacy
advertising.

Fixes #28544.

Signed-off-by: Jordan Montgomery <montytyper@msn.com>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 1]
```diff
@@ -125,35 +125,37 @@ uint8_t ll_adv_aux_ad_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref, ui
...
-		ull_hdr_init(&aux->ull);
-
-		aux->interval =	adv->interval +
-				(HAL_TICKER_TICKS_TO_US(ULL_ADV_RANDOM_DELAY) /
-				 625U);
-
-		ticks_anchor = ticker_ticks_now_get();
-
-		ticks_slot_overhead = ull_adv_aux_evt_init(aux);
+	if(adv->lll.aux && adv->is_enabled) {
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:128]:
```suggestion
	if (adv->is_enabled && adv->lll.aux) {
```

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 2]
```diff
@@ -125,35 +125,37 @@ uint8_t ll_adv_aux_ad_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref, ui
...
+		struct ll_adv_aux_set *aux = (void *)HDR_LLL2EVT(adv->lll.aux);
+		if (adv->is_enabled && !aux->is_started) {
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:130]:
```suggestion
		struct ll_adv_aux_set *aux;
		
		 aux = (void *)HDR_LLL2EVT(adv->lll.aux);
		if (!aux->is_started) {
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by cvinayak** [comment]
Also, please address the checkpatch errors too.

```
$ ./scripts/checkpatch.pl -g HEAD
```

**Review by kruithofa — APPROVED** [review]
LGTM

**Review by pabigot — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**